### PR TITLE
HiFi Calendar V2

### PR DESCRIPTION
**Pull request by MarkBrosche**

### CODE REVIEW

File: usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js [hunk 1]
```diff
@@ -7,30 +7,45 @@
...
+    var MeetingZone = function() {  
+        _this = this;     
     };
 
     MeetingZone.prototype = {
-        preload: function(entityId) {
-            _this.entityId = entityId;
-            roomName = Entities.getEntityProperties(_this.entityId, ['name']).name;
-            occupantTextEntityId = Entities.findEntitiesByName(roomName + '_OCCUPANTS', ORIGIN, RADIUS)[0];
+        preload: function(entityID) {
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js]: on preload, you need to check if the avatar that loads this script is standing within the boundaries of the zone, and add them to the list of users inside the zone if so.

i've written an example of a function that does this below
```
    var HALF = 0.5;
    function positionIsInsideEntityBounds(entityID, targetPosition) {
        targetPosition = targetPosition || MyAvatar.position;

        var properties = Entities.getEntityProperties(entityID, ["position", "dimensions", "rotation"]);
        var entityPosition = properties.position;
        var entityDimensions = properties.dimensions;
        var entityRotation = properties.rotation;
        
        var worldOffset = Vec3.subtract(targetPosition, entityPosition);
        targetPosition = Vec3.multiplyQbyV(Quat.inverse(entityRotation), worldOffset);

        var minX = -entityDimensions.x * HALF;
        var maxX = entityDimensions.x * HALF;
        var minY = -entityDimensions.y * HALF;
        var maxY = entityDimensions.y * HALF;
        var minZ = -entityDimensions.z * HALF;
        var maxZ = entityDimensions.z * HALF;

        return (targetPosition.x >= minX && targetPosition.x <= maxX
            && targetPosition.y >= minY && targetPosition.y <= maxY
            && targetPosition.z >= minZ && targetPosition.z <= maxZ);
    }
```

File: usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js [hunk 2]
```diff
@@ -7,30 +7,45 @@
...
+            if (_this.userData.length !== 0) {
+                try {
+                    _this.userData = JSON.parse(_this.userData);
+                } catch (e) {
+                    console.log(e, "Could not parse userData");
+                    return;
+                }
+                _this.occupantsListID = _this.userData.occupantsListID;
+            } else {
+                console.log("No userData found");
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js]: in this case, you want to `return`, right? (not that there's anything below this, but for security for adding code later)

File: usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js [hunk 3]
```diff
@@ -7,30 +7,45 @@
...
+            }
         },
 
         enterEntity: function() {
-            Entities.callEntityServerMethod(occupantTextEntityId, "enteredMeetingZone", [MyAvatar.sessionDisplayName]);
+            Entities.callEntityServerMethod(_this.occupantsListID, "enteredMeetingZone", [MyAvatar.sessionUUID, MyAvatar.sessionDisplayName]);
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js]: wrap this in `if (_this.occupantsListID)`

File: usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js [hunk 4]
```diff
@@ -7,30 +7,45 @@
...
         },
 
         leaveEntity: function() {
-            Entities.callEntityServerMethod(occupantTextEntityId, "leftMeetingZone", [MyAvatar.sessionDisplayName]);
+            Entities.callEntityServerMethod(_this.occupantsListID, "leftMeetingZone", [MyAvatar.sessionUUID]);
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js]: wrap this in `if (_this.occupantsListID)`

File: usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js [hunk 5]
```diff
@@ -7,30 +7,45 @@
...
+        },
+
+        leaveDomain: function(uuid) {
+            Entities.callEntityServerMethod(_this.occupantsListID, "leftMeetingZone", [uuid]);
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js]: wrap this in `if (_this.occupantsListID)`

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 6]
```diff
@@ -8,119 +8,348 @@
 
 
 (function() {
+    var INTERVAL_FREQUENCY_MS = 60000;
+    var EXPIRY_BUFFER = 300000;
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: what is `EXPIRY_BUFFER` ?

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 7]
```diff
@@ -8,119 +8,348 @@
...
+    var HOURS_PER_DAY = 24;
+    var NOON = 12;
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: units please!

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 8]
```diff
@@ -8,119 +8,348 @@
...
+    var CHANNEL = "HiFi.Google.Calendar";
+    var GREEN_TEXT = [168, 255, 168];
+    var RED_TEXT = [255, 168, 168];
+    var GREEN = [125, 255, 125];
+    var RED = [255, 125, 125];
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: could you make this and the above vars slightly more descriptive, please? maybe `SIGN_BACKGROUND_COLOR_INUSE` ?

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 9]
```diff
@@ -8,119 +8,348 @@
...
+            try {
+                that.userData = JSON.parse(that.entityProperties.userData);
+            } catch (e) {
+                console.log(e, "Could not parse userData");
+                return;
+            }
+            that.roomScheduleID = that.userData.roomScheduleID; 
+            that.roomOccupantListID = that.userData.roomOccupantListID; 
+        } else {
+            console.log("Please enter appropriate userData to enable functionality of this server script.");
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: in this error case, it doesn't look like you're stopping the functions below from executing. if that's true, please add such protection

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 10]
```diff
@@ -8,119 +8,348 @@
...
         that.entityID = entityID;
-        that.entityProperties = Entities.getEntityProperties(that.entityID, ['id', 'name', 'type', 'parentID']);
-        that.room = {};
-        if (that.entityProperties.name.indexOf("_SCHEDULE") === -1 && 
-            that.entityProperties.name.indexOf("_OCCUPANTS") === -1 && 
-            that.entityProperties.type === "Text") {
+        that.entityProperties = Entities.getEntityProperties(that.entityID, ['userData']);
+        if (that.entityProperties.userData.length !== 0) {
+            try {
+                that.userData = JSON.parse(that.entityProperties.userData);
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: why can't you do `var userData` instead of `that.userData` ? i don't think you want to be making `userData` an object inside `that`, do you?

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 11]
```diff
@@ -8,119 +8,348 @@
...
+        if (that.entityID === id) {
+            if (that.interval) {
+                console.log("CLEARING INTERVAL");
+                Script.clearInterval(that.interval);
+                that.interval = false;
+            }
+            that.entityProperties = Entities.getEntityProperties(that.entityID, ['userData']);
+            if (that.entityProperties.userData.length !== 0) {
+                try {
+                    that.userData = JSON.parse(that.entityProperties.userData);
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: why can't you do `var userData` instead of `that.userData` ? i don't think you want to be making `userData` an object inside `that`, do you?

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 12]
```diff
@@ -8,119 +8,348 @@
...
+            that.calendarID = that.userData.roomCalendarAddress;
+            that.roomColorID = that.userData.roomColorID;
+            that.roomColorOccupantsID = that.userData.roomColorOccupantsID; 
+            that.roomEntityIDs = [
+                that.roomScheduleID,
+                that.roomColorID,
+                that.roomColorOccupantsID,
+                that.roomOccupantListID
+            ];                
+            that.room = {};
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: this line is unnecessary right?

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 13]
```diff
@@ -129,38 +358,59 @@
...
             });
         }
     };  
     
     
     this.updateSignColor = function(id, params) {
-        if (that.entityID === id) {    
-            if (params[1] === "GREEN") {
+        if (id === that.roomEntityIDs[1]) {    
+            if (params[0] === "GREEN") {
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: please don't parse your colors by matching strings contained within a single-value array argument to this function.

the best way i can see to do this would be to have `updateSignColor` take a bool like `isOccupied` as its second parameter. then your conditionals could operate on the flag status instead of string matching

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 14]
```diff
@@ -8,119 +8,348 @@
...
+            "/events?&timeMin=" + (new Date()).toISOString() +
+            "&timeMax=" + tomorrowMidnight.toISOString() +
+            "&showDeleted=" + false + 
+            "&singleEvents=" + true +
+            "&maxResults=2" +
+            "&orderBy=startTime" +
+            "&access_token=" + token;
+        if (!that.interval) {
+            that.interval = Script.setInterval(function() {
+                if ((new Date()).valueOf() > (that.expireTime - EXPIRY_BUFFER) && !that.sentAlready) {
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: it looks like you are duplicating this code down on line 169? please pull both instances of this code into a new function, and have the interval call that function upon expiry

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 15]
```diff
@@ -8,119 +8,348 @@
...
+                    that.sentAlready = true;
+                    console.log("SENDING MESSAGE FOR NEW TOKEN");
+                    Messages.sendMessage(CHANNEL, JSON.stringify({
+                        type: "REFRESH TOKEN",
+                        token: that.token,
+                        uuid: that.roomScheduleID
+                    }));
+                } else if ((new Date()).valueOf() > (that.expireTime)) {
+                    console.log("EXCEEDED EXPIRATION OF TOKEN");
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: can you not ask for a refresh token in this case?

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 16]
```diff
@@ -8,119 +8,348 @@
...
+                    Script.clearInterval(that.interval);
+                    that.interval = false;
+                    return;
+                }
+                that.request(that.scheduleURL, function (error, response) {
+                    if (error) {
+                        console.log("could not complete request", error);
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: how can the script recover in this case?

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 17]
```diff
@@ -8,119 +8,348 @@
...
+                        that.clearEventList(that.entityID);
+                        console.log(JSON.stringify(response));
+                        var events = response.items;
+                        if (events.length > 0) {
+                            for (var i = 0; i < events.length; i++) {
+                                var event = events[i];
+                                var start = event.start.dateTime;
+                                var end = event.end.dateTime;
+                                var summary = event.summary;
+                                if (!start) {
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: why is this necessary?

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 18]
```diff
@@ -8,119 +8,348 @@
...
+    this.showNoScheduledEvents = function(id) {
         if (id === that.entityID) {
-            Entities.editEntity(that.room.scheduleEntityID, {text: "No events scheduled for this room"});
+            Entities.editEntity(id, {text: "Nothing on the schedule for this room today."});
         }
     };
 
 
-    this.addEvent = function(id, params) {
+    this.postEvents = function(id, params) {
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: why does this function take an array as its second argument instead of listing out the three arguments within that array? (`summary, startTimestamp, endTimestamp`)

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 19]
```diff
@@ -8,119 +8,348 @@
...
+        var msec = 0;
+        var tzHour = +m[7];
+        var tzMin = +m[8];
+        var tzOffset = new Date().getTimezoneOffset() + tzHour * MINS_PER_HOUR + tzMin;
+        var date = new Date(Date.UTC(year, month - 1, day, hour, minute - tzOffset, second, msec));
+        return date;
+    };
+
+
+    this.showNoScheduledEvents = function(id) {
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: this is also a general question: why is this function not global (instead of right now where it's a member of `this`)? and why do you need the check at line 237?

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js (hunk 18 above)
Comment (zfox23) [reply usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: also, same question as above - why is this function not global? i'm asking myself this question a few times throughout the code

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 20]
```diff
@@ -8,119 +8,348 @@
...
             };
 
             that.room.eventList.push(tempEvent);
 
-            var printedSchedule = "Today's Meetings:\n";
+            var printedSchedule = '';
             that.room.eventList.forEach(function(event) {
+                var startHours;
+                var endHours;
+                if (event.startTimestamp.getHours() - that.timezoneOffset <= 0) {
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: man, it's really too bad that we can't use ECMAScript 2015+ :(

Comment (zfox23) [reply usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: nice date parsing function though :)

File: usefulUtilities/hiFiCalendar/resources/modules/request.js [hunk 21]
```diff
@@ -0,0 +1,83 @@
+"use strict";
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/modules/request.js]: please delete this file and its containing folder :)

File: usefulUtilities/hiFiCalendar/resources/test.html [hunk 22]
```diff
@@ -0,0 +1,152 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <title>Google OAuth API Quickstart</title>
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/test.html]: please rename this html page!

File: usefulUtilities/hiFiCalendar/resources/test.html [hunk 23]
```diff
@@ -0,0 +1,152 @@
...
+  <meta charset="utf-8" />
+</head>
+<body>
+    <p>Google OAuth API Quickstart</p>
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/test.html]: necessary?

File: usefulUtilities/hiFiCalendar/resources/test.html [hunk 24]
```diff
@@ -0,0 +1,152 @@
...
+                'clientId': CLIENT_ID,
+                'scope': SCOPE
+            }).then(function () {
+                GoogleAuth = gapi.auth2.getAuthInstance();
+        
+                // Listen for sign-in state changes.
+                GoogleAuth.isSignedIn.listen(updateSigninStatus);
+                GoogleAuth.grantOfflineAccess().then(function (response) {
+                    authCode = response.code;
+                    var zone = new Date().toLocaleTimeString('en-us',{timeZoneName:'short'}).split(' ')[2];
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/test.html]: this seems very fragile, is there no other way to get this data?

File: usefulUtilities/hiFiCalendar/resources/test.html [hunk 25]
```diff
@@ -0,0 +1,152 @@
...
+                });
+                // Handle initial sign-in state. (Determine if user is already signed in.)
+                updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());
+                var user = GoogleAuth.currentUser.get();
+                setSigninStatus();
+        
+                // Call handleAuthClick function when user clicks on
+                //      "Sign In/Authorize" button.
+                $('#sign-in-or-out-button').click(function() {
+                handleAuthClick();
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/test.html]: add tab

File: usefulUtilities/hiFiCalendar/resources/test.html [hunk 26]
```diff
@@ -0,0 +1,152 @@
...
+                }); 
+                $('#revoke-access-button').click(function() {
+                revokeAccess();
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/test.html]: add tab

File: usefulUtilities/hiFiCalendar/resources/test.html [hunk 27]
```diff
@@ -0,0 +1,152 @@
...
+            var isAuthorized = user.hasGrantedScopes(SCOPE);
+
+            if (isAuthorized) {
+                $('#sign-in-or-out-button').html('Sign out');
+                $('#revoke-access-button').css('display', 'inline-block');
+                $('#auth-status').html('You are currently signed in and have granted ' +
+                    'access to this app.');
+                EventBridge.emitWebEvent(JSON.stringify({
+                    type: "TOKEN",
+                    token: response.access_token,
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/test.html]: where is `response` coming from?

File: usefulUtilities/hiFiCalendar/resources/test.html [hunk 28]
```diff
@@ -0,0 +1,152 @@
...
+        }
+
+
+        // This function receives and attempts to parse data from 
+        // High Fidelity scripts.
+        function onScriptEventReceived(data) {
+            var data = JSON.parse(data);
+            switch (data.type) {
+                case "RENEW_TOKEN":
+                updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/resources/test.html]: add tab

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js [hunk 29]
```diff
@@ -8,119 +8,348 @@
...
-                "id": that.entityProperties.id, 
-                "name": that.entityProperties.name,
-                "eventList": [],
-                "scheduleEntityID": that.entityProperties.parentID
+                "eventList": []
             };
-            Entities.editEntity(that.room.scheduleEntityID, {text: ""});
-            that.updateSignColor(that.room.id, [that.room.name, "GREEN"]);
-        } else {
+            that.request = Script.require('https://hifi-content.s3.amazonaws.com/brosche/dev/googleCalendar/resources/modules/request.js').request;
```
[the remark this answers is not in the document]
Comment (MarkBrosche) [reply usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: Can server scripts access the request module without at least a relative link?  I was thinking I would use Script.resolvePath('../resources/modules/request.js').

[the remark this answers is not in the document]
Comment (zfox23) [reply usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: I'm not sure. Even if the answer is yes, I don't want to add another copy of `request.js` to our repo.

File: usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js (hunk 1 above)
Comment (MarkBrosche) [reply usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js]: When I reload the script with people in the room they get picked up, so I'm not sure this is necessary.

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js (hunk 6 above)
Comment (MarkBrosche) [reply usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: It is the window of time before the token expires (5 minutes) in which the server should make it's request to refresh the token.

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js (hunk 9 above)
Comment (MarkBrosche) [reply usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: Isn't calling "return" sufficient protection?  We don't want the script to run if the userData is incomplete or incorrect.

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js (hunk 10 above)
Comment (MarkBrosche) [reply usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: I was concerned that creating var userData would affect other entities running this script.

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js (hunk 11 above)
Comment (MarkBrosche) [reply usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: see above response.

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js (hunk 16 above)
Comment (MarkBrosche) [reply usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: it cannot! user intervention is required in this case.

File: usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js (hunk 17 above)
Comment (MarkBrosche) [reply usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js]: This is from the example google calendar script, I'm guessing it handles the case where somebody hasn't entered exact time information, say for an all day event.  It'd be interesting to see if this can actually handle something like that.

File: usefulUtilities/hiFiCalendar/resources/test.html (hunk 24 above)
Comment (MarkBrosche) [reply usefulUtilities/hiFiCalendar/resources/test.html]: Are you asking about getting the timezone name? this is the only way I found.

File: usefulUtilities/hiFiCalendar/entities/meetingRoomCalendar.json [hunk 30]
```diff
@@ -0,0 +1,186 @@
...
+        "x": 0.0001678466796875,
+        "y": -0.0000152587890625,
+        "z": 0.0000762939453125,
+        "w": 1
+    },
+    "grab": {
+        "grabbable": false
+    },
+    "collisionless": true,
+    "serverScripts": "http://hifi-content.s3-us-west-1.amazonaws.com/rebecca/LoadTest/clockPST.js",
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entities/meetingRoomCalendar.json]: what's this?

File: usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js [hunk 31]
```diff
@@ -30,22 +31,54 @@
                 _this.occupantsListID = _this.userData.occupantsListID;
             } else {
                 console.log("No userData found");
+                return;
+            }
+            if (_this.positionIsInsideEntityBounds(_this.entityID, MyAvatar.position)) {
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js]: yeahhhh!!

File: usefulUtilities/hiFiCalendar/README.md [hunk 32]
```diff
@@ -30,5 +30,28 @@ This app lets you integrate with the Google Calendar API to query meeting room s
 - Initial Release
 
 ## TO DO
-1. Remove Room Label Entity and just have a schedule entity that changes colors in next version
-2. Refactor authorization to allow app to run in the background.
\ No newline at end of file
+1. Simplify setup experience.
+
+## Technical Details:
+Here's what I'd expect here:
```
Comment (zfox23) [inline usefulUtilities/hiFiCalendar/README.md]: is all of this accurate? if it is, you don't need this line. if it isn't, please make it so :)

File: usefulUtilities/hiFiCalendar/entities/meetingRoomCalendar.json (hunk 30 above)
Comment (MarkBrosche) [reply usefulUtilities/hiFiCalendar/entities/meetingRoomCalendar.json]: This is the clock at the top of the schedules... I didn't realize this wasn't released code... I will investigate.
